{lib}mediainfo{-gui}: 0.7.82 -> 0.7.83

devhell opened the pull request:
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [x] Built on platform(s): NixOS
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Built and tested locally.

From the changelog:

``````
Version 0.7.83, 2016-02-29

  + HEVC: Maximum Content Light Level (MaxCLL) and Maximum Frame-Average
     Light Level (MaxFALL), metadata mandated by CEA-861.3 for HDR support
  + HEVC: Mastering display color primaries and luminance (based on SMPTE
    ST 2084), metadata mandated by CEA-861.3 for HDR support
  + HEVC: SMPTE ST 2048 and SMPTE ST 428-1 transfer characteristics
  + HEVC: Chroma subsampling location (indication of the location type
    described in the HEVC spec)
  + MPEG-TS: ATSC Modulation Mode (Analog, SCTE_mode_1 aka 64-QAM,
    SCTE_mode_2 aka 256-QAM, 8-VSB, 16-VSB)
  + #B981, MP4: support of buggy file having "hint" SubType
  x HLS: better handling of media playlists having EXT-X-BYTERANGE
    ```
``````

aneeshusa commented:
Could you also please add a commit to update the `src.url` and `meta.homepage` attributes to use HTTPS for these packages as well? Refs #13809.

devhell commented:
@aneeshusa Done.

hrdinka commented on pkgs/applications/misc/mediainfo-gui/default.nix:
```diff
@@ -1,11 +1,11 @@
+    sha256 = "0d8mph9lbg2lw0ccg1la0kqhbisra8q9rzn195lncch5cia5zyg7";
   };
 
   buildInputs = [ automake autoconf libtool pkgconfig libzen libmediainfo wxGTK desktop_file_utils libSM imagemagick ];
```
Derivations like `automake` or `autoconf` are used by the build system and not by the resulting application. These are needed in the building machines native architecture and therefore belong into `nativeBuildInputs`. Think of a x86 host (native gcc, make) building for arm (arm libc, arm libz).

devhell committed:
{lib}mediainfo{-gui}: 0.7.82 -> 0.7.83

Built and tested locally.

From the changelog:

```
Version 0.7.83, 2016-02-29

+ HEVC: Maximum Content Light Level (MaxCLL) and Maximum Frame-Average
  Light Level (MaxFALL), metadata mandated by CEA-861.3 for HDR support
+ HEVC: Mastering display color primaries and luminance (based on SMPTE
  ST 2084), metadata mandated by CEA-861.3 for HDR support
+ HEVC: SMPTE ST 2048 and SMPTE ST 428-1 transfer characteristics
+ HEVC: Chroma subsampling location (indication of the location type
  described in the HEVC spec)
+ MPEG-TS: ATSC Modulation Mode (Analog, SCTE_mode_1 aka 64-QAM,
  SCTE_mode_2 aka 256-QAM, 8-VSB, 16-VSB)
+ #B981, MP4: support of buggy file having "hint" SubType
x HLS: better handling of media playlists having EXT-X-BYTERANGE
```

Additionally, some cleanup and package uses `autoreconfHook` now. Thanks
to @hrdinka for helpful pointers!

hrdinka commented on pkgs/applications/misc/mediainfo-gui/default.nix:
```diff
@@ -1,28 +1,28 @@
 
-  buildInputs = [ automake autoconf libtool pkgconfig libzen libmediainfo wxGTK desktop_file_utils libSM imagemagick ];
+  nativeBuildInputs = [ autoreconfHook pkgconfig libzen libmediainfo wxGTK
+                        desktop_file_utils libSM imagemagick ];
```
Oh. I think you misunderstood me here. `autoreconfHook` `pkgconfig` are `nativeBuildInputs` the remaining are `buildInputs`. The difference isn't document well, here's a example:
Assume we are cross building an application for ARM on a x86 host, this means we need the following:
`gcc` is needed on the host machine -> x86
`pkgconfig`, `make`, `autoconf` are run on the host as well and never needed on the target machine -> x86
`zlib` `wxGtk`.. are build & runtime dependencies for the app, these are needed for ARM architecture and are never run on the host

To tell our build system whats needed in which architecture we place build tools in `nativeBuildInputs` (= native to the building machine) and application dependencies in `buildInputs` (= native to target architecture).

devhell replied:
That was a really excellent explanation. Thank you very much! :)

hrdinka commented on pkgs/development/libraries/libmediainfo/default.nix:
```diff
@@ -1,29 +1,28 @@
   };
 
-  buildInputs = [ automake autoconf libtool pkgconfig libzen zlib ];
+  nativeBuildInputs = [ autoreconfHook pkgconfig libzen zlib ];
```
You've missed correcting it here.

devhell replied:
Sorry about that. Hope it's okay now.

hrdinka commented:
Built & tested locally (NixOS x86_64). Thanks for your contribution! I highly appreciate your detailed commit messages I will adopt this for my future commits.
